app/gui2/vite.config.ts

```diff
@@ -94,13 +100,48 @@ export default defineConfig({
   },
 })
 
+let ydocServer: ChildProcessWithoutNullStreams | null
 function gatewayServer(): Plugin {
   return {
     name: 'gateway-server',
     configureServer(server) {
       if (server.httpServer == null) return
 
-      createGatewayServer(server.httpServer, undefined)
+      if (IS_POLYGLOT_YDOC_SERVER) {
+        const ydocServerJar = fileURLToPath(
+          new URL(
+            '../../lib/java/ydoc-server/target/ydoc-server-assembly-0.1.0-SNAPSHOT.jar',
+            import.meta.url,
+          ),
+        )
+        const runYdocServer = () => {
+          const args = []
+          if (IS_POLYGLOT_YDOC_SERVER_DEBUG) {
+            args.push('-DinspectPort=34567')
+          }
+          args.push('-jar', ydocServerJar)
+          ydocServer = spawn('java', args)
+          if (IS_POLYGLOT_YDOC_SERVER_DEBUG) {
+            ydocServer.stdout.on('data', (data) => console.log(`ydoc: ${data}`))
+          }
+          ydocServer.stderr.on('data', (data) => console.log(`ydoc: ${data}`))
+        }
+        if (!existsSync(ydocServerJar)) {
+          const cwd = fileURLToPath(new URL('../..', import.meta.url))
+          const sbt = spawn('sbt', ['ydoc-server/assembly'], { cwd })
```

Comment:
That seems like an unnecessary dependency. If someone forgot to build the jar, tough luck.